rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py

```diff
@@ -153,14 +158,17 @@ def _run_task(self, task_definition):
             response_dict = ecs_client.run_task(**task_definition)
             if response_dict['failures']:
                 logger.error("Failures were found: [{}]".format(response_dict['failures']))
-                status_id = StatusConstants.FAILED
+                if 'limit' in response_dict['failures'][0]['reason']:
+                    raise ECSLimitReached(response_dict['failures'][0]['reason'])
+                else:
+                    status_id = StatusConstants.FAILED
             elif not response_dict['tasks']:
                 logger.error("No tasks were returned")
                 status_id = StatusConstants.READY
             elif response_dict['tasks']:
                 for task in response_dict['tasks']:
                     assigned_to = '--ecs--{}'.format(task['taskArn'])
-        except (KeyError, ClientError) as exception:
+        except (KeyError, ClientError, IndexError) as exception:
```

Comment:
you might get index error if the structure of the response isn't what you expect? in which case, it's just an unknown failure, like the "else" above. 👍

Reply:
Yep. That is right.